src/state/state.js

```diff
@@ -188,14 +188,14 @@
          * @ignore
          */
         function _renderFrame(time) {
-            // update all game objects
-            me.game.update(time);
-            // render all game objects
-            me.game.draw();
             // schedule the next frame update
             if (_animFrameId !== -1) {
                 _animFrameId = window.requestAnimationFrame(_renderFrame);
             }
+            // update all game objects
+            me.game.update(time);
+            // render all game objects
+            me.game.draw();
```

Comment:
Regardless of the magic the browser creates with requestAnimationFrame, this is an actual issue when using setTimeout.

Here's the case:
Call setTimeout (16ms) -> Code executes for (15ms) -> setTimeout resolves 1ms later -> draw!
Code executes for (15ms) -> call setTimeout (16ms) -> setTimeout resolves 16ms later -> missed a frame!